catalyst/core/runner.py

```diff
@@ -154,7 +154,8 @@ def _get_experiment_components(
     def _prepare_for_stage(self, stage: str):
         utils.set_global_seed(self.experiment.initial_seed)
         migrating_params = {}
-        if self.state is not None:
+        migrate = self.experiment.get_state_params(stage).get("migrate", True)
+        if self.state is not None and migrate:
```

Comment:
could you please create something like `stage_state_params`, and reuse them during State creation? I just don't like the idea for several `self.experiment.get_state_params` call, if there is no need in it